Fix EEPROM errors when EXTRUDERS == 0

vector76 opened the pull request:
### Description

This is a fix for an error (partly mine) on a previous PR that fixed EEPROM for EXTRUDERS == 0.  (https://github.com/MarlinFirmware/Marlin/pull/16464)

Previously, the MIN(EXTRUDERS, 1) was a bonehead mistake that would have broken multi-extruders but had no meaningful effect for EXTRUDERS == 0.  A subsequent change to the more 'sane' MAX(EXTRUDERS, 1) ended up breaking EXTRUDERS == 0.

The problem in its current state appears to be that forcing store/retrieve of one element (even when EXTRUDERS == 0) is inconsistent with offsetof() calculation based on the size of planner_extruder_advance_K which is zero.  Instead, use save/retrieve of zero elements. This is natural for the saving (remove _MAX()), no special consideration needed.  For the retrieval, a simple #ifdef covers it.  Reading a zero-sized object is not equivalent to suppressing the reading altogether, which is why this is needed.

### Benefits

EEPROM data size error no longer occurs when EXTRUDERS == 0.

### Related Issues

Related to PR https://github.com/MarlinFirmware/Marlin/pull/16464

vector76 committed:
Fix EEPROM errors when EXTRUDER == 0

Partly my fault from last time (https://github.com/MarlinFirmware/Marlin/pull/16464)
Problem appears to be that forcing store/retrieve of one element is inconsistent with offsetof() calculation based on size of planner_extruder_advance_K which is zero.  Instead, use save/retrieve of zero elements, which is natural for the saving (remove _MAX()) and for the retrieval a simple #ifdef covers it.

thinkyhead commented:
It's a custom in Marlin to write a placeholder to EEPROM for certain disabled features, so the correct solution is to apply `_MAX()` in the data structure.

vector76 commented:
Ok, tonight I will make another attempt that conforms with this convention.  I think it might entail forcing a minimum size for planner_extruder_advance_K.

vector76 committed:
Storage space for one value instead of suppressing

Remove suppression of load/store of extruder advance K.
Instead, allocate space for one value
Without this, it fails on EEPROM size when EXTRUDERS == 0

vector76 commented:
Ok, backed out the change and fixed a different way: _MAX(EXTRUDERS, 1) on planner_extruder_advance_K was all that was needed.  Tested and it no longer reports EEPROM data size error.

Failed CI tests appear to be unrelated.

thinkyhead committed:
Merge remote-tracking branch 'upstream/bugfix-2.0.x' into bf2_eeprom_patch_16898

thinkyhead committed:
Merge remote-tracking branch 'upstream/bugfix-2.0.x' into bf2_eeprom_patch_16898
